svc.go

```diff
@@ -96,6 +105,28 @@ func (as *AutoNATService) handleStream(s network.Stream) {
 	}
 }
 
+// Optimistically extract the net.IP host from a multiaddress.
+func addrToIP(addr ma.Multiaddr) (net.IP, error) {
```

Comment:
This looks like a common function and should probably be put in `go-multiaddr-net ` ?